### PR TITLE
Fix up various TLS and config issues

### PR DESCRIPTION
**Pull request by dbason**
* Creates certs for the bootstrap pod
 * Mounts volumes to the bootstrap pod
 * Adds discovery service name to cert SANs
 * Mounts volumes to securityconfig job
 * Belts and braces testing for all of the above
 * Update Dockerfile with new files

### CODE REVIEW

**Review by idanl21 — COMMENTED** [review]
Hey dan , looks good.
when im clone it and try to create an Dcoker image out of it I'm getting a lot of errors
` Failed to watch *v1.Pod: failed to list *v1.Pod: pods is forbidden: User "system:serviceaccount:opensearch-operator-system:opensearch-operator-controller-manager" cannot list resource "pods" in API group "" at the cluster scope0`
Do you have a clue ?

File: opensearch-operator/pkg/reconcilers/tls.go [hunk 1]
```diff
@@ -223,6 +224,28 @@ func (r *TLSReconciler) handleTransportGeneratePerNode() error {
 	}
 	nodeSecret.Data[CaCertKey] = ca.CertData()
 
+	// Generate bootstrap pod cert
+	bootstrapPodName := builders.BootstrapPodName(r.instance)
+	dnsNames := []string{
```
[the remark this answers is not in the document]
Comment (dbason) [reply opensearch-operator/pkg/reconcilers/tls.go:229]: Yeah correct, need to add transport certs for bootstrap pod

File: opensearch-operator/Dockerfile [hunk 2]
```diff
@@ -23,6 +23,7 @@ RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
 FROM gcr.io/distroless/static:nonroot
 WORKDIR /
+COPY helperfiles/   helperfiles/
```
[the remark this answers is not in the document]
Comment (dbason) [reply opensearch-operator/Dockerfile:26]: I think if we include those files we should embed them as then they're available for unit tests as well, and it's a bit cleaner.  But I'm still not convinced we should include those files.  I've got #99 open, and I've got some code sitting in a branch to resolve this if we want, but I wanted to wait until everyone agreed.

**Review by idanl21 — APPROVED** [review]
approved

**Pull request merged**